src/hammer-vlsi/utils_test.py

```diff
@@ -49,6 +51,44 @@ def test_optional_map(self) -> None:
         self.assertNotEqual(optional_map("88", str_to_num), "880")
         self.assertEqual(optional_map("42", str_to_num), 420)
 
+    def test_coerce_to_grid(self) -> None:
+        self.assertEqual(coerce_to_grid(1.23, Decimal("0.1")), Decimal("1.2"))
+        self.assertEqual(coerce_to_grid(1.23, Decimal("0.01")), Decimal("1.23"))
+        self.assertEqual(coerce_to_grid(1.227, Decimal("0.01")), Decimal("1.23"))
+        self.assertEqual(coerce_to_grid(200, Decimal("10")), Decimal("200"))
```

Comment:
I think we should be setting the decimal precision. Add then add a test for decimals out of precision i.e. "0.0001"

Reply:
Where would the precision live? Another variable? IMO the `grid_unit` _is_ the precision.

Reply:
So you want to set the precision to `len(grid_unit)` or something like that?

Reply:
I don't think the precision is doing quite what you're (and I was) expecting. It's for decimal-to-decimal math, not float-to-decimal conversion. For example, it doesn't solve the Decimal(0.001) problem:
```
>>> getcontext().prec = 3
>>> Decimal(0.001)
Decimal('0.001000000000000000020816681711721685132943093776702880859375')
>>> Decimal("0.001") # 1 sig fig
Decimal('0.001')
>>> Decimal("1.00001") # 6 sig figs
Decimal('1.00001')
>>> Decimal("0.001") + Decimal("1.00001") # will result in prec=3 sig figs
Decimal('1.00')
```

Reply:
Ok well at least we tried.